### PR TITLE
tracker/tracker2: add tests and featureflag

### PR DESCRIPTION
**Pull request by dB2510**
Adds tests for tracker2 and featureflag with status alpha.

category: feature
ticket: #1478 
feature_flag: tracker_v2

### CODE REVIEW

File: core/tracker/tracker2/tracker.go [hunk 1]
```diff
@@ -166,6 +166,10 @@ const (
 	// This indicates a bug in charon as it is unexpected.
 	msgFetcherSyncContributionFailedSigAggPrepare = "couldn't fetch sync contribution due to no aggregated sync contribution selection, this is unexpected"
 
+	// msgFetcherUnknownError indicates duty failed in fetcher step with some unknown error.
+	// This indicates a bug in charon as it is unexpected.
+	msgFetcherUnknownError = "couldn't fetch due to unknown error"
```
Comment (corverroos) [inline core/tracker/tracker2/tracker.go:171]: can improve this to check for eth2 errors

Comment (dB2510) [reply core/tracker/tracker2/tracker.go:173]: done

File: core/tracker/tracker2/tracker.go [hunk 2]
```diff
@@ -407,7 +417,8 @@ func analyseDutyFailed(duty core.Duty, allEvents map[core.Duty][]event, msgRootC
 // and the reason which might actually be due a pre-requisite duty that failed.
 func analyseFetcherFailed(duty core.Duty, allEvents map[core.Duty][]event, fetchErr error) string {
 	// Check for beacon api errors.
-	if strings.Contains(fetchErr.Error(), "beacon api") {
+	var eth2Error eth2http.Error
+	if errors.As(fetchErr, &eth2Error) {
```
Comment (dB2510) [inline core/tracker/tracker2/tracker.go:421]: @corverroos here we check for eth2 errors

File: core/tracker/tracker2/tracker_internal_test.go [hunk 3]
```diff
@@ -0,0 +1,1129 @@
...
+
+	tests := []struct {
+		name   string
+		duty   core.Duty
+		events map[core.Duty][]event
+		msg    string
+		failed bool
+	}{
+		{
+			name: "eth2 error",
```
Comment (dB2510) [inline core/tracker/tracker2/tracker_internal_test.go:711]: here we test for eth2 error

File: core/tracker/tracker2/tracker.go [hunk 4]
```diff
@@ -166,6 +167,10 @@ const (
 	// This indicates a bug in charon as it is unexpected.
 	msgFetcherSyncContributionFailedSigAggPrepare = "couldn't fetch sync contribution due to no aggregated sync contribution selection, this is unexpected"
 
+	// msgFetcherUnknownError indicates duty failed in fetcher step with some unknown error.
+	// This indicates a bug in charon as it is unexpected.
+	msgFetcherUnknownError = "couldn't fetch due to unknown error"
```
Comment (corverroos) [inline core/tracker/tracker2/tracker.go:172]: `couldn't fetch due to unexpected error`

File: core/interfaces.go [hunk 5]
```diff
@@ -187,12 +187,18 @@ type Tracker interface {
...
+
 	// ParSigDBStoredExternal sends ParSigDB component's store external events to tracker.
 	ParSigDBStoredExternal(context.Context, Duty, ParSignedDataSet, error)
 
-	// SigAggAggregated sends SigAgg component's events to tracker.
+	// SigAggAggregated sends SigAgg component's aggregate events to tracker.
 	SigAggAggregated(context.Context, Duty, PubKey, []ParSignedData, error)
 
+	// AggSigSBStored sends AggSigDB component's store events to tracker.
+	AggSigSBStored(context.Context, Duty, PubKey, SignedData, error)
```
Comment (corverroos) [inline core/interfaces.go:200]: typo

File: core/tracker/tracker2/tracker.go [hunk 6]
```diff
@@ -37,11 +37,14 @@ const (
 	dutyDB                 // Duty data stored in DutyDB
 	validatorAPI           // Partial signed data from local VC submitted to vapi
 	parSigDBInternal       // Partial signed data from local VC stored in parsigdb
-	parSigEx               // Partial signed data from other VC received via parsigex
+	parSigExBroadcast      // Partial signed data from local VC broadcasted to other peers in cluster
+	parSigExReceive        // Partial signed data from other VC received via parsigex
```
Comment (corverroos) [inline core/tracker/tracker2/tracker.go:41]: remove this

File: core/tracker/tracker2/tracker.go [hunk 7]
```diff
@@ -94,6 +100,11 @@ const (
 	// This indicates the associated prepare aggregation duty failed.
 	msgFetcherAggregatorFailedPrepare = "couldn't aggregate attestation due to failed prepare aggregator duty"
 
+	// msgFetcherAggregatorFailedPrepareBroadcast indicates an attestation aggregation duty failed in
+	// the fetcher step since it couldn't fetch the prerequisite aggregated beacon committee selections.
+	// This indicates the associated prepare aggregation duty failed due to failure in broadcast of partial beacon committee selections to other peers.
+	msgFetcherAggregatorFailedPrepareBroadcast = "couldn't aggregate attestation due to failed broadcast of partial beacon committee selections to other peers"
+
```
[the remark this answers is not in the document]
Comment (corverroos) [reply core/tracker/tracker2/tracker.go:107]: also it will never happen since parsigex.Broadcast never actually returns an error

**Review by corverroos — APPROVED** [review]

**Pull request merged**